File: client/javascript-closure-angular/API/Client/QueryList.js
goog.provide('API.Client.QueryList');

/**
 * @record
 */
API.Client.QueryList = function() {}

/**
 * @type {!API.Client.Metadata}
 * @export
 */
API.Client.QueryList.prototype.metadata;

/**
 * @type {!API.Client.Pagination}
 * @export
 */
API.Client.QueryList.prototype.pagination;

/**
 * @type {!Array<!API.Client.Query>}
 * @export
 */
API.Client.QueryList.prototype.data;

File: client/scalaz/src/main/scala/org/openapitools/client/api/QueryList.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import QueryList._

case class QueryList (
  metadata: Option[Metadata],
pagination: Option[Pagination],
data: Option[List[Query]])

object QueryList {
  import DateTimeCodecs._

  implicit val QueryListCodecJson: CodecJson[QueryList] = CodecJson.derive[QueryList]
  implicit val QueryListDecoder: EntityDecoder[QueryList] = jsonOf[QueryList]
  implicit val QueryListEncoder: EntityEncoder[QueryList] = jsonEncoderOf[QueryList]
}
